---
bool success
string message
int32[] board_state
